ios/samples/hello-gltf/hello-gltf/FilamentView/FilamentApp.cpp

```diff
@@ -0,0 +1,167 @@
+    engine->destroy(scene);
+    engine->destroy(view);
+    engine->destroy(camera);
+    engine->destroy(swapChain);
+    engine->destroy(&engine);
+}
+
+void FilamentApp::setupFilament() {
+#if FILAMENT_APP_USE_OPENGL
+    engine = Engine::create(filament::Engine::Backend::OPENGL);
+#elif FILAMENT_APP_USE_METAL
+    engine = Engine::create(filament::Engine::Backend::METAL);
+#endif
+    swapChain = engine->createSwapChain(nativeLayer);
+    renderer = engine->createRenderer();
+    scene = engine->createScene();
+    camera = engine->createCamera();
+    cameraManipulator.setCamera(camera);
+    cameraManipulator.setViewport(width, height);
+    cameraManipulator.lookAt(filament::math::double3(0, 0, 3), filament::math::double3(0, 0, 0));
+    camera->setProjection(60, (float) width / height, 0.1, 10);
+}
+
+void FilamentApp::setupIbl() {
+    image::KtxBundle* iblBundle = new image::KtxBundle(RESOURCES_VENETIAN_CROSSROADS_IBL_DATA,
+                                                       RESOURCES_VENETIAN_CROSSROADS_IBL_SIZE);
+    float3 harmonics[9];
+    parseSphereHarmonics(iblBundle->getMetadata("sh"), harmonics);
+    app.iblTexture = image::KtxUtility::createTexture(engine, iblBundle, false, true);
+
+    image::KtxBundle* skyboxBundle = new image::KtxBundle(RESOURCES_VENETIAN_CROSSROADS_SKYBOX_DATA,
+                                                          RESOURCES_VENETIAN_CROSSROADS_SKYBOX_SIZE);
+    app.skyboxTexture = image::KtxUtility::createTexture(engine, skyboxBundle, false, true);
+
+    app.skybox = Skybox::Builder()
+        .environment(app.skyboxTexture)
+        .build(*engine);
+
+    app.indirectLight = IndirectLight::Builder()
+        .reflections(app.iblTexture)
+        .irradiance(3, harmonics)
+        .intensity(30000)
+        .build(*engine);
+    scene->setIndirectLight(app.indirectLight);
+    scene->setSkybox(app.skybox);
+
+    app.sun = EntityManager::get().create();
+    LightManager::Builder(LightManager::Type::SUN)
+        .castShadows(true)
+        .direction({0.0, -1.0, 0.0})
+        .build(*engine, app.sun);
+    scene->addEntity(app.sun);
+}
+
+void FilamentApp::setupMesh() {
+    app.materialProvider = gltfio::createUbershaderLoader(engine);
+    app.assetLoader = gltfio::AssetLoader::create({engine, app.materialProvider, nullptr});
+
+    // Load the glTF file.
+    std::ifstream in(resourcePath.concat(utils::Path("DamagedHelmet.glb")), std::ifstream::in);
+    in.seekg(0, std::ios::end);
+    std::ifstream::pos_type size = in.tellg();
+    in.seekg(0);
+    std::vector<uint8_t> buffer(size);
+    if (!in.read((char*) buffer.data(), size)) {
+        std::cerr << "Unable to read scene.gltf" << std::endl;
+        exit(1);
+    }
+    app.asset = app.assetLoader->createAssetFromBinary(buffer.data(), static_cast<uint32_t>(size));
+
+    gltfio::ResourceLoader({
+        .engine = engine,
+        .gltfPath = resourcePath.concat(utils::Path("DamagedHelmet.glb")),
```

Comment:
Can this line be removed? If not, we perhaps have a bug in gltfio.

Reply:
Yup, tested and it works.